src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/sealedshouldhavepermitslist/InputSealedShouldHavePermitsListInnerClass.java

```diff
@@ -0,0 +1,21 @@
+/*
+SealedShouldHavePermitsList
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.design.sealedshouldhavepermitslist;
+
+public class InputSealedShouldHavePermitsListInnerClass {
+     sealed class A {} // violation
+     final class B extends A {}
+     final class C extends A {}
+     final class D { } // this can extend A, so as any other class in the compilation unit
+}
+
+class corrected {
```

Comment:
Please always follow common Java paradigms to allow us to focus on what is important here, please do for all:

```suggestion
class InputSealedShouldHavePermitsListInnerClassCorrect {
```